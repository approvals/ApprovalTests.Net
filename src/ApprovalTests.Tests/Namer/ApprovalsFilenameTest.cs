public class ApprovalsFilenameTest
{
    [Test]
    public void TestMachineSpecificName()
    {
        // begin-snippet: approvals_filename
        var approvalsFilename = ApprovalsFilename.Parse(@"..\Email\EmailTest.Testname.Microsoft_Windows_10_Education.approved.eml");
        // end-snippet
        Approvals.Verify(approvalsFilename);
        ClassicAssert.True(approvalsFilename.IsMachineSpecific);
    }

    [Test]
    public void TestNonMachineSpecificName() =>
        Approvals.Verify(ApprovalsFilename.Parse(@"..\Email\EmailTest.Testname.approved.eml"));

    [Test]
    public void TestSimilarFiles()
    {
        var file = PathUtilities.GetAdjacentFile(@"..\Email\EmailTest.Testname.Microsoft_Windows_10_Education.approved.eml");
        var approvalsFilename = ApprovalsFilename.Parse(file);
        Approvals.VerifyAll("Like EmailTest.Testname.Microsoft_Windows_10_Education.approved.eml", approvalsFilename.GetOtherMachineSpecificFiles(), f => f.Name);
    }
}