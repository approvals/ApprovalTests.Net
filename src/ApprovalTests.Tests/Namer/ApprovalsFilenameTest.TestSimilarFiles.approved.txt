Like EmailTest.Testname.Microsoft_Windows_10_Education.approved.eml

EmailTest.Testname.Microsoft_Windows_10_Home_N.approved.eml
EmailTest.Testname.Microsoft_Windows_10_Pro.approved.eml
EmailTest.Testname.Microsoft_Windows_Server_2016_Datacenter.approved.eml
EmailTest.Testname.Microsoft_Windows_Server_2019_Datacenter.approved.eml
